### DISCUSSION TITLE
How to add a new colorscheme?

### DISCUSSION THREAD

**Discussion by thieung** [fitrh/init.nvim#4] [Q&A] [answered]
Hi @fitrh 
Thanks for your awesome config, I'm using it :)

In case I want to add new colorscheme, could you please help to write some notes/guides for it?
I'm not still understand how it works in your config.

Thank you!

**Comment by fitrh** [comment, marked as answer]
> In case I want to add new colorscheme, could you please help to write some notes/guides for it?

That's a good idea, for now, you can follow these steps:

1. Add the new theme packer specification to `lua/config/plugin/colorscheme/packerspec.lua` and set the `as` key to `colorscheme/{theme name}`
2. Create `lua/config/plugin/colorscheme/{theme name}/setup.lua`, this is where you initialize the theme like setting the configuration, calling the `vim.cmd.colorscheme`, or override the theme
3. Add your `{theme name}` to `after/plugin/colorscheme.lua` in the `colorscehmes` table with the key and value as same as `{theme name}`
4. Now, you can use the theme by setting `NVIM_COLORSCHEME` environment variable to `{theme name}`, you can do that by exporting the variable or using environment overrides
    - Exporting environment variable
    
      ```sh
      # in bash, zsh or POSIX-compilant shell
      # You can put it in your shell init file or just run it before running nvim
      export NVIM_COLORSCHEME="kanagawa"
      ```

      ```fish
      # in fish shell
      set -x NVIM_COLORSCHEME doom-one
      ```
  
    - Environment overrides
    
      ```sh
      NVIM_COLORSCHEME=gruvbox nvim
      ```

You can use the available themes as a reference

- [after/plugin/colorscheme.lua](https://github.com/fitrh/init.nvim/blob/main/after/plugin/colorscheme.lua)
- [lua/config/plugin/colorscheme](https://github.com/fitrh/init.nvim/tree/main/lua/config/plugin/colorscheme)

Since c4136e96112410af9b7273a99839a1a4841e752e, you can control theme variants using `NVIM_{THEME NAME}_VARIANT` and/or `NVIM_{THEME NAME}_MODE` environment variables

```fish
# in fish shell
set -x NVIM_COLORSCHEME material
set -x NVIM_MATERIAL_VARIANT palenight
nvim # launch neovim with marko-cerovac/material.nvim using its palenight variant
set -x NVIM_MATERIAL_VARIANT "deep ocean"
nvim # now with material deep ocean variant
set -x NVIM_COLORSCHEME gruvbox
set -x NVIM_GRUVBOX_VARIANT hard
nvim # with gruvbox using the "hard" contrast
set -x NVIM_GRUVBOX_MODE light
nvim # gruvbox hard in light mode
```

Each theme has a default value if the environment variable is not set.

It might seem complicated, but for me, it is easier to switch themes or set different themes for each session or even machine just by setting environment variable without needing to change the configuration code.

FYI, I use 4 window managers (and actively switch between them) and each one has a different theme by using the appropriate environment variables in their startx/init files.

**Comment by thieung** [comment]
Thanks so much for your detail guide.
Could you please check "doom-one" colorscheme, seem like it doesn't work.

**Reply by fitrh** [reply]
`doom-one` is the most recently added colors scheme, maybe you are not using the latest commit(?)

**Reply by thieung** [reply]
I cloned your repo and installed on my machine for reference, but looks like it does not work. Am I missing something?

**Reply by fitrh** [reply]
Is running `NVIM_COLORSCHEME="doom-one" nvim` not working? Try `git pull` first to ensure that you are on the latest commit

**Reply by thieung** [reply]
![image](https://user-images.githubusercontent.com/79964592/188424181-be832ee3-fede-47f2-9202-5e68b97b7960.png)
Does it look like this @fitrh ?

**Reply by fitrh** [reply]
Yes, that what `doom-one` is like, isn't it? `NVIM_COLORSCHEME="doom-one" NVIM_DOOM_ONE_MODE="light" nvim` should give you the light mode

**Reply by thieung** [reply]
Thanks @fitrh , I checked again and it works on your config, hmm, let me check on my own again :)
